code/android-signal-library/src/main/java/com/adobe/marketing/mobile/signal/SignalHitProcessor.kt

```diff
@@ -42,18 +36,19 @@ internal class SignalHitProcessor : HitProcessing {
         return HIT_QUEUE_RETRY_TIME_SECONDS
     }
 
-    override fun processHit(entity: DataEntity?): Boolean {
-        if (entity == null) {
-            Log.warning(SignalConstants.LOG_TAG, CLASS_NAME,"Drop this data entity as it is null.")
-            return true
+    override fun processHit(entity: DataEntity, processingResult: HitProcessingResult) {
+        if (entity == null || processingResult == null) {
+            Log.warning(SignalConstants.LOG_TAG, CLASS_NAME, "Drop this data entity as it is null.")
```

Comment:
You missed calling completion handler here

Reply:
`processingResult` could be null here, we shouldb't call the completion handler. And I don't think we need to add more logic to do it when the `entity` is null, and the `processingResult` is not null. 
It's a sanity check, the code should never get there since the arguments should be @NonNull.

Reply:
We can remove these null checks here, as the SDK always sends a non-null param. You can also annotate these params `@Nonnull`. It is confusing when reading code as returning without calling completion handler blocks the entire Hit processor.